README.md

```diff
@@ -355,6 +356,38 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React props & TypeScript
+
+`interface` is preferred over `type` when defining the component props. Are a little more powerful
+than the corresponding type declaration, but for a set of react props, it likely doesn't matter.
+
+Also is not necessary to type the `returnType` because is inferred from what you actually return, so
+can be omitted most of the time.
+
+So the best approach and example would be something like this:
+
+```
+interface FancyButtonProps {
+  color: string
+}
+
+const FancyButton = (props: FancyButtonProps) => {
```

Comment:
I would like the opinion of others, but I think that writing components using `function` instead of an arrow function is better. As with our lint rules, there's no way that you can not name a `function`, avoiding unnamed elements in React's dev tools. As a bonus, you can also immediately `export default` that component.

```ts
export default function MyComponent(props) {}
```

Reply:
I'm with you on that. Somehow everyone (including myself) started to write arrow functions a lot, because it looked new and fancy.
IMO there are mostly downsides to writing top-level arrow functions, like the fact that they don't hoist (so you can't call them before you declared them) among other things.

One of the only upsides of arrow functions is that you can "type" them by typing the resulting variable, and have everything in there (incl args and return types) inferred. However, even that has its challenges, e.g. when giving arguments default values that might not match the prop type or something like that.